Sources/Features/Breadcrumb/BacktraceBreadcrumbs.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+
+@objc public enum BacktraceBreadcrumbType: Int {
+
+    case manual = 1
+    case log = 2
+    case navigation = 3
+    case http = 4
+    case system = 5
+    case user = 6
+    case configuration = 7
+
+    var description: String {
+        switch self {
+        case .manual:
+            return "manual"
+        case .log:
+            return "log"
+        case .navigation:
+            return "navigation"
+        case .http:
+            return "http"
+        case .system:
+            return "system"
+        case .user:
+            return "user"
+        case .configuration:
+            return "configuration"
+        }
+    }
+
+    public static let all: [BacktraceBreadcrumbType] = [.manual, .log, .navigation, .http, .system, .user, .configuration]
+
+    public static let none: [BacktraceBreadcrumbType] = []
+}
+
+@objc public enum BacktraceBreadcrumbLevel: Int {
+
+    case debug = 1
+    case info = 2
+    case warning = 3
+    case error = 4
+    case fatal = 5
+
+    var description: String {
+        switch self {
+        case .debug:
+            return "debug"
+        case .info:
+            return "info"
+        case .warning:
+            return "warning"
+        case .error:
+            return "error"
+        case .fatal:
+            return "fatal"
+        }
+    }
+}
+
+@objc public class BacktraceBreadcrumbs: NSObject {
+
+    private var breadcrumbsLogManager: BacktraceBreadcrumbsLogManager?
+    private var backtraceNotificationObserver: BacktraceNotificationObserver?
+    private var breadcrumbLevel: BacktraceBreadcrumbLevel?
+    private var breadcrumbTypes: [BacktraceBreadcrumbType]?
+    private(set) var isBreadcrumbsEnabled: Bool = false
+
+    public func enableBreadcrumbs(_ breadcrumbSettings: BacktraceBreadcrumbSettings = BacktraceBreadcrumbSettings()) {
+        do {
+            self.breadcrumbLevel = breadcrumbSettings.breadcrumbLevel
+            self.breadcrumbTypes = breadcrumbSettings.breadcrumbTypes
+            self.breadcrumbsLogManager = try BacktraceBreadcrumbsLogManager(breadcrumbSettings: breadcrumbSettings)
+            if breadcrumbSettings.breadcrumbTypes.contains(where: { $0.rawValue == BacktraceBreadcrumbType.system.rawValue }) {
+                backtraceNotificationObserver = BacktraceNotificationObserver(breadcrumbs: self)
+                backtraceNotificationObserver?.enableNotificationObserver()
```

Comment:
is there a chance that the backtraceNotificationObserver will be nullable?

Reply:
On line 76, it's pretty much guaranteed to not be null

Reply:
It could be:
```
let backtraceNotificationObserver = BacktraceNotificationObserver(breadcrumbs: self) backtraceNotificationObserver.enableNotificationObserver()
self.backtraceNotificationObserver = backtraceNotificationObserver
```
I prefer the current code.

WDYT?